pkg/build/controller/factory/factory.go

```diff
@@ -218,16 +218,22 @@ type typeBasedFactoryStrategy struct {
 }
 
 func (f *typeBasedFactoryStrategy) CreateBuildPod(build *buildapi.Build) (*kapi.Pod, error) {
+	var pod *kapi.Pod
+	var err error
 	switch build.Parameters.Strategy.Type {
 	case buildapi.DockerBuildStrategyType:
-		return f.DockerBuildStrategy.CreateBuildPod(build)
+		pod, err = f.DockerBuildStrategy.CreateBuildPod(build)
 	case buildapi.STIBuildStrategyType:
-		return f.STIBuildStrategy.CreateBuildPod(build)
+		pod, err = f.STIBuildStrategy.CreateBuildPod(build)
 	case buildapi.CustomBuildStrategyType:
-		return f.CustomBuildStrategy.CreateBuildPod(build)
+		pod, err = f.CustomBuildStrategy.CreateBuildPod(build)
 	default:
 		return nil, errors.New("No strategy defined for type")
 	}
+	if pod != nil {
```

Comment:
i debated.  the previous logic returned both args returned from CreateBuildPod, so i was trying to retain that.  Whether i should attempt to set the annotation when doing so or not is debatable.   I figured it's possible "err" is non-fatal and the caller is still going to create the pod, in which case we do want to add the annotation.